sources/Algo/include/Algo.h

```diff
@@ -192,13 +192,24 @@ class GeneratorDefinitionAlgorithm : public NodeAlgorithm {
   /**
    * @brief Perform algorithm
    *
-   * Add the generators of the nodes and deducing the model to use
+   * Add the generators of the nodes and deducing the model to use. If the node is a generator with a deduced
+   * model of a diagram then the function isDiagramValid is called. If the diagram is not valid the generator
```

Comment:
"If the diagram is not valid the generator..." => If the diagram is not valid the default generator model is used

Reply:
incorrect. from algorithm point of view, if diagram is invalid, the generator is ignored

Reply:
both are equivalent I guess

Reply:
Let's keep it that way as the warning will give the consequence